alf/drivers/on_policy_driver.py

```diff
@@ -227,12 +227,12 @@ def create_ta(s):
                  name='iter_loop')
 
         if self._final_step_mode == OnPolicyDriver.FINAL_STEP_SKIP:
-            next_time_step, policy_step, action = self._step(
+            next_time_step, policy_step, action, time_step = self._step(
                 time_step, policy_state)
             next_state = policy_step.state
         else:
+            time_step = self._algorithm.transform_timestep(time_step)
             policy_step = common.algorithm_step(self._algorithm.rollout,
-                                                self._observation_transformer,
                                                 time_step, policy_state)
             action = common.sample_action_distribution(policy_step.action)
             next_time_step = time_step
```

Comment:
For the two different branches, `next_time_step` is not consistently transformed.  I believe that nontransformed `next_time_step` should be returned.

Reply:
good catch